trajopt_sco/src/optimizers.cpp

```diff
@@ -271,11 +222,272 @@ struct MultiCritFilter {
+                  approx_improve,
+                  exact_improve,
+                  "  ------  ");
+  }
+
+  if (cnt_names.size() != 0)
+  {
+    std::printf("%15s | %10s---%10s---%10s---%10s\n",
+                "CONSTRAINTS",
+                "----------",
+                "----------",
+                "----------",
+                "---------"
+                "-");
+    for (size_t i = 0; i < old_cnt_viols.size(); ++i)
+    {
+      double approx_improve = old_cnt_viols[i] - model_cnt_viols[i];
+      double exact_improve = old_cnt_viols[i] - new_cnt_viols[i];
+      if (fabs(approx_improve) > 1e-8)
+        std::printf("%15s | %10.3e | %10.3e | %10.3e | %10.3e\n",
+                    cnt_names[i].c_str(),
+                    merit_error_coeff * old_cnt_viols[i],
+                    merit_error_coeff * approx_improve,
+                    merit_error_coeff * exact_improve,
+                    exact_improve / approx_improve);
+      else
+        std::printf("%15s | %10.3e | %10.3e | %10.3e | %10s\n",
+                    cnt_names[i].c_str(),
+                    merit_error_coeff * old_cnt_viols[i],
+                    merit_error_coeff * approx_improve,
+                    merit_error_coeff * exact_improve,
+                    "  ------  ");
+    }
+  }
+
+  std::printf("%15s | %10.3e | %10.3e | %10.3e | %10.3e\n",
+              "TOTAL",
+              old_merit,
+              approx_merit_improve,
+              exact_merit_improve,
+              merit_improve_ratio);
+}
+
+void BasicTrustRegionSQPResults::writeSolver(std::FILE* stream, bool header) const
+{
+  if (header)
+    std::fprintf(stream, "%s,%s,%s,%s,%s\n", "DESCRIPTION", "oldexact", "dapprox", "dexact", "ratio");
+
+  std::fprintf(stream,
+               "%s,%10.3e,%10.3e,%10.3e,%10.3e\n",
+               "Solver",
+               old_merit,
+               approx_merit_improve,
+               exact_merit_improve,
+               merit_improve_ratio);
+  std::fflush(stream);
+}
+
+void BasicTrustRegionSQPResults::writeVars(std::FILE* stream, bool header) const
+{
+  if (header)
+  {
+    std::fprintf(stream, "%s", "NAMES");
+    for (const auto& var : var_names)
+      std::fprintf(stream, ",%s", var.c_str());
+
+    std::fprintf(stream, "\n");
+  }
+
+  std::fprintf(stream, "%s", "VALUES");
+  for (const auto& x : new_x)
+    std::fprintf(stream, ",%e", x);
+
+  std::fprintf(stream, "\n");
+  std::fflush(stream);
+}
+
+void BasicTrustRegionSQPResults::writeCosts(std::FILE* stream, bool header) const
+{
+  if (header)
+  {
+    std::fprintf(stream, "%s", "COST NAMES");
+    for (const auto& name : cost_names)
+      std::fprintf(stream, ",%s,%s,%s,%s", name.c_str(), name.c_str(), name.c_str(), name.c_str());
+
+    std::fprintf(stream, "\n");
+
+    std::fprintf(stream, "%s", "DESCRIPTION");
+    for (size_t i = 0; i < cost_names.size(); ++i)
+      std::fprintf(stream, ",%s,%s,%s,%s\n", "oldexact", "dapprox", "dexact", "ratio");
```

Comment:
drop the \n here. Same in constraints

```
std::fprintf(stream, ",%s,%s,%s,%s", "oldexact", "dapprox", "dexact", "ratio");
```

Reply:
> @mpowelson I made the requested changes. Are the changes to how the log file is written correct?

See the one comment. When I made that change it looked fine.

Reply:
Should be fixed now.